### LOAD DATASET 

In [19]:
import pandas as pd
import numpy as np
SuperStore= pd.read_csv("C:/Users/ayoub/Desktop/janvier-2027/python w2/projet/data/Superstore21.csv" ,sep=',' ,encoding='latin1' )
SuperStore

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
9989,9990,CA-2014-110422,1/21/2014,1/23/2014,Second Class,TB-21400,Tom Boeckenhauer,Consumer,United States,Miami,...,33180,South,FUR-FU-10001889,Furniture,Furnishings,Ultra Door Pull Handle,25.2480,3,0.20,4.1028
9990,9991,CA-2017-121258,2/26/2017,3/3/2017,Standard Class,DB-13060,Dave Brooks,Consumer,United States,Costa Mesa,...,92627,West,FUR-FU-10000747,Furniture,Furnishings,Tenex B1-RE Series Chair Mats for Low Pile Car...,91.9600,2,0.00,15.6332
9991,9992,CA-2017-121258,2/26/2017,3/3/2017,Standard Class,DB-13060,Dave Brooks,Consumer,United States,Costa Mesa,...,92627,West,TEC-PH-10003645,Technology,Phones,Aastra 57i VoIP phone,258.5760,2,0.20,19.3932
9992,9993,CA-2017-121258,2/26/2017,3/3/2017,Standard Class,DB-13060,Dave Brooks,Consumer,United States,Costa Mesa,...,92627,West,OFF-PA-10004041,Office Supplies,Paper,"It's Hot Message Books with Stickers, 2 3/4"" x 5""",29.6000,4,0.00,13.3200


### PRINTING DATASET INFORMATIONS 


In [20]:
print("the number of rows is :" , SuperStore.shape[0] )
print("the number of columns is :" , SuperStore.shape[1] )
print( SuperStore.info(memory_usage=False))

the number of rows is : 9994
the number of columns is : 21
<class 'pandas.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   str    
 2   Order Date     9994 non-null   str    
 3   Ship Date      9994 non-null   str    
 4   Ship Mode      9994 non-null   str    
 5   Customer ID    9994 non-null   str    
 6   Customer Name  9994 non-null   str    
 7   Segment        9994 non-null   str    
 8   Country        9994 non-null   str    
 9   City           9994 non-null   str    
 10  State          9994 non-null   str    
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   str    
 13  Product ID     9994 non-null   str    
 14  Category       9994 non-null   str    
 15  Sub-Category   9994 non-null   str    
 16  Product Name   9994 non-null   str    
 17  Sale

### First and last  5 elements 


In [21]:
print(SuperStore.head())
print("------")
print(SuperStore.tail())

   Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   Sout

#### Columns to be changed
We can change ship date from str to a datetime object  rather than a string
and we can change the postal code into an integer rather than float

### STATS

In [22]:
print(SuperStore.describe())

            Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000  


### Missing value

In [23]:
print(SuperStore.isnull().sum())
print("the percentage of missing value in postal code is :" , ((SuperStore["Postal Code"].isna().sum()) / len(SuperStore)) * 100 ,"%")
Missing=SuperStore[SuperStore["Postal Code"].isna()][["City","State" , "Postal Code"]]
print(Missing)

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64
the percentage of missing value in postal code is : 0.0 %
Empty DataFrame
Columns: [City, State, Postal Code]
Index: []


As we can see from the data all postal code that are missing are for vermont state so it is suitable to fill the nan values with Burligtone vermont postal code which is 05401

In [24]:
SuperStore["Postal Code"] = SuperStore["Postal Code"].fillna(5401.0)
print(SuperStore.isnull().sum())
SuperStore["Postal Code"]=SuperStore["Postal Code"].astype(int).astype(str).str.zfill(5)
print("the new type of Postal code is " , SuperStore["Postal Code"].dtype)


Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64
the new type of Postal code is  str


### Duplicated values

In [25]:
ColumnsToCheck=SuperStore.columns.drop('Row ID')
SuperStore[SuperStore.duplicated(subset=ColumnsToCheck, keep=False)]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
3405,3406,US-2014-150119,4/23/2014,4/27/2014,Standard Class,LB-16795,Laurel Beltran,Home Office,United States,Columbus,...,43229,East,FUR-CH-10002965,Furniture,Chairs,Global Leather Highback Executive Chair with P...,281.372,2,0.3,-12.0588
3406,3407,US-2014-150119,4/23/2014,4/27/2014,Standard Class,LB-16795,Laurel Beltran,Home Office,United States,Columbus,...,43229,East,FUR-CH-10002965,Furniture,Chairs,Global Leather Highback Executive Chair with P...,281.372,2,0.3,-12.0588


because of teh Row ID value we cant have a duplicated row because it increment from 1 to 9800 however if we check the dataset without the Row ID we indeed find a repeated row 
but it is really difficult to distinguish if it is a repeated order or it is an actuall duplicate 

for our case we will consider it as repeated becaus eit is very  very unlikely that a person commit two same orders in the same day and in an addition they are consecutive in the dataset Row id = 3406 , 3407 

In [26]:
DropIndex=SuperStore[SuperStore["Row ID"] == 3407].index
SuperStore.drop(DropIndex , inplace=True)
SuperStore[SuperStore.duplicated(subset=ColumnsToCheck, keep=False)]


,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit


### Data consistency

#### categorical values inconsistency 
Ship mode : we have 4 unique values  ['Second Class', 'Standard Class', 'First Class', 'Same Day'] all of these values are consistente and have meaning  

Segment :  we have 3 unique values ['Consumer', 'Corporate', 'Home Office'] this is consistent

Country : we have 1 unique values ['United States'] this is valid

State : to be considered to be consistente and because we have one country covered by this data which is america we should expect no more than 50 state which is true 

Region : ['South', 'West', 'Central', 'East']are  valid

category : ['Furniture', 'Office Supplies', 'Technology']  3 product category which are valid

Sub Category : we crossed checked category with sub category to find if there is somme inconsstencies such ass categori: fourniter-sub category = Arts which we didnot find 

thats make us conclude that our data set categorical values are consistente 

In [27]:
print(SuperStore["Ship Mode"].unique())
print(SuperStore["Segment"].unique())
print(SuperStore["Country"].unique())
print(SuperStore["State"].nunique())
print(SuperStore["Region"].unique())
print(SuperStore[SuperStore["Category"] == "Furniture"]["Sub-Category"].unique())
print(SuperStore[SuperStore["Category"] == "Office Supplies"]["Sub-Category"].unique())
print(SuperStore[SuperStore["Category"] == "Technology"]["Sub-Category"].unique())


<StringArray>
['Second Class', 'Standard Class', 'First Class', 'Same Day']
Length: 4, dtype: str
<StringArray>
['Consumer', 'Corporate', 'Home Office']
Length: 3, dtype: str
<StringArray>
['United States']
Length: 1, dtype: str
49
<StringArray>
['South', 'West', 'Central', 'East']
Length: 4, dtype: str
<StringArray>
['Bookcases', 'Chairs', 'Tables', 'Furnishings']
Length: 4, dtype: str
<StringArray>
[    'Labels',    'Storage',        'Art',    'Binders', 'Appliances',
      'Paper',  'Envelopes',  'Fasteners',   'Supplies']
Length: 9, dtype: str
<StringArray>
['Phones', 'Accessories', 'Machines', 'Copiers']
Length: 4, dtype: str


#### Cross data consistency
Order ID : for the same order we can not find different Order IDs which is true in our Data Set

Customer ID Customer Name : we must have the same number of Customer ID and Customer Name wxhich is true 

Product ID and Product Name :  we must have the same number of Product ID and   Product Name which is False  we have more ditinct Product ID than Distinc Product Name which means there are some product which have different IDs


In [28]:
ColumnsToCheck2=SuperStore.columns.drop(['Row ID' , 'Order ID'])
print("The number of duplicated orders is " , SuperStore.duplicated(subset=ColumnsToCheck2, keep=False).sum())
print(SuperStore["Customer ID"].nunique() == SuperStore["Customer Name"].nunique() )
print(SuperStore["Product ID"].nunique(),SuperStore["Product Name"].nunique() )

CountN=SuperStore.groupby("Product ID")["Product Name"].nunique().reset_index()
IDsNames=CountN[CountN["Product Name"]>1]
print(IDsNames)



The number of duplicated orders is  0
True
1862 1850
           Product ID  Product Name
18    FUR-BO-10002213             2
65    FUR-CH-10001146             2
183   FUR-FU-10001473             2
284   FUR-FU-10004017             2
290   FUR-FU-10004091             2
295   FUR-FU-10004270             2
310   FUR-FU-10004848             2
311   FUR-FU-10004864             2
387   OFF-AP-10000576             2
506   OFF-AR-10001149             2
721   OFF-BI-10002026             2
831   OFF-BI-10004632             2
832   OFF-BI-10004654             2
1044  OFF-PA-10000357             2
1049  OFF-PA-10000477             2
1064  OFF-PA-10000659             2
1086  OFF-PA-10001166             2
1144  OFF-PA-10001970             2
1156  OFF-PA-10002195             2
1168  OFF-PA-10002377             2
1198  OFF-PA-10003022             2
1331  OFF-ST-10001228             2
1418  OFF-ST-10004950             2
1517  TEC-AC-10002049             2
1533  TEC-AC-10002550             2
1569  TEC-A

as we can see here we have 60 row which have a probleme in their product Name we will standrize them for every product id x we will assigne for it the one product name y and we cant tolerate that our solution is based on the issue because if a product id has muiltiple names it is a really big probleme

In [29]:
ProblemIDs= IDsNames["Product ID"]

TargetMap = SuperStore[SuperStore['Product ID'].isin(ProblemIDs)].groupby("Product ID")["Product Name"].first().to_dict()
SuperStore.loc[SuperStore["Product ID"].isin(ProblemIDs) , "Product Name"]=SuperStore["Product ID"].map(TargetMap)

SuperStore["Product ID"].nunique()-SuperStore["Product Name"].nunique()

44

as we can see the result is -32 that means that we have a probleme in the other direction one product name has multiple Ids attached to it 
the probleme is that we kan give a product name multiple id because it might have the same name a chair but the color is different for example ared chair has id1 and a black chair has id2 which is plausible with a one name to elemnate the confusion we decided to create a new column contains a unique name  that solves the probleme 

In [30]:
SuperStore["Unique Product Name"]=SuperStore["Product Name"]+"("+SuperStore["Product ID"]+")"
SuperStore["Product ID"].nunique()-SuperStore["Unique Product Name"].nunique()

0

at this point our dataset is valid the number of names equales the number of IDs for all the products 

#### Negative or zero value 
we do not have any zero or sub zero values in the numerical colomns 


In [31]:
print(SuperStore[SuperStore["Row ID"] <= 0])
print(SuperStore[SuperStore["Sales"] <= 0])
print((SuperStore["Postal Code"] == "0").sum() ) 

Empty DataFrame
Columns: [Row ID, Order ID, Order Date, Ship Date, Ship Mode, Customer ID, Customer Name, Segment, Country, City, State, Postal Code, Region, Product ID, Category, Sub-Category, Product Name, Sales, Quantity, Discount, Profit, Unique Product Name]
Index: []

[0 rows x 22 columns]
Empty DataFrame
Columns: [Row ID, Order ID, Order Date, Ship Date, Ship Mode, Customer ID, Customer Name, Segment, Country, City, State, Postal Code, Region, Product ID, Category, Sub-Category, Product Name, Sales, Quantity, Discount, Profit, Unique Product Name]
Index: []

[0 rows x 22 columns]
0


#### Date inconsistencies 
we will turn the dates from strings to datetime object so the manipulation is easiere 

In [32]:
SuperStore["Order Date"]=pd.to_datetime(SuperStore["Order Date"] , errors="coerce" )
SuperStore["Ship Date"]=pd.to_datetime(SuperStore["Ship Date"] , errors="coerce" )

print(SuperStore["Order Date"].isna().sum())
print(SuperStore["Ship Date"].isna().sum())


0
0


Now we check for logical inconsistencies such as 

shippin date must always be > than order Date 

minimum and maximum values

In [33]:
TimeO=SuperStore[SuperStore["Ship Date"] < SuperStore ["Order Date"]]
print((TimeO))
print(SuperStore["Ship Date"].min())
print(SuperStore["Ship Date"].max())
print(SuperStore["Order Date"].min())
print(SuperStore["Order Date"].max())

Empty DataFrame
Columns: [Row ID, Order ID, Order Date, Ship Date, Ship Mode, Customer ID, Customer Name, Segment, Country, City, State, Postal Code, Region, Product ID, Category, Sub-Category, Product Name, Sales, Quantity, Discount, Profit, Unique Product Name]
Index: []

[0 rows x 22 columns]
2014-01-07 00:00:00
2018-01-05 00:00:00
2014-01-03 00:00:00
2017-12-30 00:00:00


In [34]:
SuperStore.info()

<class 'pandas.DataFrame'>
Index: 9993 entries, 0 to 9993
Data columns (total 22 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   Row ID               9993 non-null   int64         
 1   Order ID             9993 non-null   str           
 2   Order Date           9993 non-null   datetime64[us]
 3   Ship Date            9993 non-null   datetime64[us]
 4   Ship Mode            9993 non-null   str           
 5   Customer ID          9993 non-null   str           
 6   Customer Name        9993 non-null   str           
 7   Segment              9993 non-null   str           
 8   Country              9993 non-null   str           
 9   City                 9993 non-null   str           
 10  State                9993 non-null   str           
 11  Postal Code          9993 non-null   str           
 12  Region               9993 non-null   str           
 13  Product ID           9993 non-null   str         

### Data anlysis

#### With pandas


In [35]:
SuperStore.describe()

,Row ID,Order Date,Ship Date,Sales,Quantity,Discount,Profit
count,9993.000000,9993,9993,9993.000000,9993.000000,9993.000000,9993.000000
mean,4997.659161,2016-04-30 01:53:33.089162,2016-05-04 00:53:19.039327,229.852846,3.789753,0.156188,28.660971
min,1.000000,2014-01-03 00:00:00,2014-01-07 00:00:00,0.444000,1.000000,0.000000,-6599.978000
25%,2499.000000,2015-05-23 00:00:00,2015-05-27 00:00:00,17.280000,2.000000,0.000000,1.731000
50%,4998.000000,2016-06-26 00:00:00,2016-06-29 00:00:00,54.480000,3.000000,0.200000,8.671000
75%,7496.000000,2017-05-14 00:00:00,2017-05-18 00:00:00,209.940000,5.000000,0.200000,29.364000
max,9994.000000,2017-12-30 00:00:00,2018-01-05 00:00:00,22638.480000,14.000000,0.800000,8399.976000
std,2885.264122,NaN,NaN,623.276074,2.225149,0.206457,234.271476


In [36]:
SuperStore.var(numeric_only=True)

Row ID      8.324749e+06
Sales       3.884731e+05
Quantity    4.951288e+00
Discount    4.262461e-02
Profit      5.488312e+04
dtype: float64

The average ordre sales value is : 229.852846	

the median is the value that if we order the data from smalest to largest devides the data in half which means 50 % fall behind and 50 % fall above

yes in both sales and profite the mean is significantly  bigger than the median which means that our data is right sekwed because for example the avergae ordre is 229 dollars and the median value is 54 wihc is significantly low which means that the big sales value interfers in our dataset by bringing the mean upword which is validated by the max and the min values in sales which the difference is huge 22638 and 0.44  

the numerical variable with the maximum variabilite is sales because its std and its max-min is the highest it may be because of the nature of the data sales is always bigger that the profit  we will use teh coefficient of variance to determine which of the data is th e most variable  



In [52]:
CoV1=SuperStore["Sales"].std() / SuperStore["Sales"].mean()
CoV2=SuperStore["Profit"].std() / SuperStore["Profit"].mean()
print(CoV1,CoV2)

2.7116308768323245 8.173884892456547


as we can see our analysis based on std and ranges was wrong according to the coefficient of variance Profit has more variablity

for the existence of teh outliers we will use the Interquartile method to calculate 

In [55]:
Q1=SuperStore["Sales"].quantile(0.25)
Q3=SuperStore["Sales"].quantile(0.75)
IQR=Q3-Q1

UpperBond=Q3+(1.5*IQR)
LowerBond=Q1-(1.5*IQR)
print ("upper :" , len(SuperStore[SuperStore["Sales"] > UpperBond]))
print ("upper :" , len(SuperStore[SuperStore["Sales"] < LowerBond]))

upper : 1167
upper : 0


we have 1167 values that can be considered outliers 

#### With numpy

In [50]:
Sales=np.array(SuperStore["Sales"])
print("the mean for sales is ",Sales.mean())
print("the median for sales is ",np.percentile(Sales , 50))
print("the minimum for sales is ",Sales.min())
print("the maximum for sales is ",Sales.max())
print("the standard deviation for sales is ",Sales.std())
print("the variance for sales is ",Sales.var())
print("the 25% for sales is ",np.percentile(Sales , 25))
print("the 50% for sales is ",np.percentile(Sales , 25))
print("the 75% for sales is ",np.percentile(Sales , 75))

the mean for sales is  229.8528458220755
the median for sales is  54.48
the minimum for sales is  0.444
the maximum for sales is  22638.48
the standard deviation for sales is  623.2448874450571
the variance for sales is  388434.1897264019
the 25% for sales is  17.28
the 50% for sales is  17.28
the 75% for sales is  209.94


### Sales analysis

In [128]:
print("the total sales values is  :" , SuperStore["Sales"].sum())
print("the total profit values is  :" , SuperStore["Profit"].sum())
print("the total quantity values is  :" , SuperStore["Quantity"].sum())
print("the average sales par ordre values is  :" , SuperStore["Sales"].sum() / SuperStore["Order ID"].count())
print("the average sales par ordre values is  :" , SuperStore["Profit"].sum() / SuperStore["Order ID"].count())
CategorySales=SuperStore.groupby("Category")["Sales"].sum()
print("the category that generates maximum sales values is :" ,CategorySales.idxmax() )
CategoryProfit=SuperStore.groupby("Category")["Profit"].sum()
print("the category that generates maximum Profit values is :" ,CategoryProfit.idxmax() )
SubCategorySales=SuperStore.groupby("Sub-Category")["Sales"].sum()
print("the sub category that generates maximum sales values is :" ,SubCategorySales.idxmax() )
SubCategoryProfit=SuperStore.groupby("Sub-Category")["Sales"].sum()
print("the sub category that generates maximum profit values is :" ,SubCategoryProfit.idxmax() )
ProductMax=SuperStore.groupby("Unique Product Name")["Sales"].sum()
PM=ProductMax.sort_values(ascending=False).head(1)
print("the product that generate the most sales is :" , PM.index[0])
ProductMaxP=SuperStore.groupby("Unique Product Name")["Profit"].sum()
PMP=ProductMax.sort_values(ascending=False).head(1)
print("the product that generate the most profit is :" , PMP.index[0])
SalesProfit=SuperStore.groupby("Unique Product Name")[['Sales' ,"Profit"]].sum()
SalesProfit[(SalesProfit["Sales"]>0 )& (SalesProfit["Profit"]<0)]

the total sales values is  : 2296919.4883000003
the total profit values is  : 286409.0805
the total quantity values is  : 37871
the average sales par ordre values is  : 229.8528458220755
the average sales par ordre values is  : 28.660970729510655
the category that generates maximum sales values is : Technology
the category that generates maximum Profit values is : Technology
the sub category that generates maximum sales values is : Phones
the sub category that generates maximum profit values is : Phones
the product that generate the most sales is : Canon imageCLASS 2200 Advanced Copier(TEC-CO-10004722)
the product that generate the most profit is : Canon imageCLASS 2200 Advanced Copier(TEC-CO-10004722)


,Sales,Profit
Unique Product Name,,
3.6 Cubic Foot Counter Height Office Refrigerator(OFF-AP-10002534),2946.200,-872.0752
36X48 HARDFLOOR CHAIRMAT(FUR-FU-10003247),369.248,-40.4914
"3D Systems Cube Printer, 2nd Generation, White(TEC-MA-10002073)",2339.982,-467.9964
3M Replacement Filter for Office Air Cleaner for 20' x 33' Room(OFF-AP-10004980),303.360,-20.4768
ARKON Windshield Dashboard Air Vent Car Mount Holder(TEC-PH-10000169),216.960,-15.5940
...,...,...
XtraLife ClearVue Slant-D Ring Binders by Cardinal(OFF-BI-10000285),72.912,-13.3280
Zebra GK420t Direct Thermal/Thermal Transfer Printer(TEC-MA-10001695),703.710,-938.2800
i.Sound Portable Power - 8000 mAh(TEC-PH-10003800),413.322,-58.8189
